src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/MavenUtils.cs

```diff
@@ -0,0 +1,187 @@
+/// Utilities for retrieving information from maven packages that may not be present on the buildDropPath.
+/// </summary>
+public class MavenUtils : IMavenUtils
+{
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    private static readonly string EnvHomePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "HOMEPATH" : "HOME";
+    private static readonly string HomePath = Environment.GetEnvironmentVariable(EnvHomePath);
+    private static readonly string MavenPackagesPath = $"{HomePath}/.m2/repository";
+
+    private bool MavenPackagesPathHasReadPermissions => fileSystemUtils.DirectoryHasReadPermissions(MavenPackagesPath);
+
+    private readonly string userDefinedLocalRepositoryPath;
+
+    public MavenUtils(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+
+        userDefinedLocalRepositoryPath = GetLocalRepositoryPath();
+    }
+
+    // Takes in a scanned component and attempts to find the associated pom file. If it is not found then it returns null.
+    public string GetPomLocation(ScannedComponent scannedComponent)
+    {
+        var pomLocation = MavenPackagesPath;
+
+        // Check if a user-defined local repository exists. If not, continue with the default value.
+        if (!string.IsNullOrEmpty(userDefinedLocalRepositoryPath))
+        {
+            pomLocation = userDefinedLocalRepositoryPath;
+        }
+
+        var componentName = scannedComponent.Component.PackageUrl?.Name.ToLower();
+        var componentNamespace = scannedComponent.Component.PackageUrl?.Namespace?.ToLower();
+        var componentVersion = scannedComponent.Component.PackageUrl?.Version;
+
+        // Take the component namespace and split it by "." in order to get the correct path to the .pom
+        if (!string.IsNullOrEmpty(componentNamespace))
+        {
+            var componentNamespaceParts = componentNamespace.Split('.'); // Example: "org.apache.commons.commons-lang3"
+
+            for (var i = 0; i < componentNamespaceParts.Length; i++)
+            {
+                pomLocation += $"/{componentNamespaceParts[i]}";
+            }
+        }
+
+        pomLocation += $"/{componentName}/{componentVersion}/{componentName}-{componentVersion}.pom";
+
+        // Check for file permissions on the .m2 directory before attempting to check if the file exists
+        if (MavenPackagesPathHasReadPermissions)
+        {
+            if (fileSystemUtils.FileExists(pomLocation))
+            {
+                return pomLocation;
+            }
+            else
+            {
+                log.Debug($"Pom location could not be found at: {pomLocation}");
```

Comment:
This would mean that for the specific package in question there is no pom.xml at the local package cache. This could be either because it doesn't exist or because we're it's not in the place we expect it to be. I'll change this to verbose.